src/components/Attachment/index.ts

```diff
@@ -362,12 +362,14 @@ class Attachment extends BridgeBase {
         )
         .then((fileRes) => {
           
-          this.thePConn.attachmentsInfo = {
+          const reqObj = {
             type: "File",
             attachmentFieldName: this.att_valueRef,
             category: this.att_categoryName,
-            ID: fileRes.ID    // 8.7 - ID is at the top-level now.
+            ID: fileRes.ID
           };
+
+          this.thePConn.attachmentsInfo = PCore.getPCoreVersion()?.includes('8.8') ? [reqObj] : reqObj;
```

Comment:
Does this logic work for the next versions of infinity? I think we should modify this logic to handle two cases
1. versions below 8.7
2. versions above 8.7

Reply:
updated the PR by modifying the condition to support 8.7 and above versions, for versions below 8.7 we have different branch no need to handle.